src/gc.c

```diff
@@ -308,6 +325,8 @@ static void gc_periodicCallback(RedisModuleCtx *ctx, void *privdata) {
 end:
 
   RedisModule_ThreadSafeContextUnlock(ctx);
+
+  return status == STATUS_OK;
```

Comment:
So status is always ok, even if invalid spec?

Reply:
if status in INVALID_SPEC then the function will return false and the gc thread will exit